src/main/kotlin/no/nav/tiltakspenger/saksbehandling/behandling/service/behandling/IverksettBehandlingService.kt

```diff
@@ -119,33 +121,54 @@ class IverksettBehandlingService(
         sakStatistikk: StatistikkSakDTO,
         stønadStatistikk: StatistikkStønadDTO,
     ): Sak {
-        val (oppdatertSak, meldeperioder) = this.genererMeldeperioder(clock)
-        // Denne har vi behov for å gjøre ved påfølgende førstegangsbehandligner (altså ikke den første)
-        val (oppdaterteMeldekortbehandlinger, oppdaterteMeldekort) =
-            this.meldekortBehandlinger.oppdaterMedNyeKjeder(oppdatertSak.meldeperiodeKjeder, tiltakstypeperioder, clock)
+        when (vedtak.vedtaksType) {
+            Vedtakstype.INNVILGELSE,
+            Vedtakstype.STANS,
```

Comment:
Vi ønsker ikke kjøre genererMeldeperioder hvis det er stans. Helt fair og dele denne funksjonen i 2, en for avslag og en for innvilgelse.

Reply:
neeevermind. Hodet mitt blandet stans og avslag. Men kan fint dele denne opp i flere funksjoner da.